sql/analyzer/validation_rules.go

```diff
@@ -260,6 +279,26 @@ func validateGroupBy(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (s
 	return n, nil
 }
 
+func expressionReferencesAlias(expr sql.Expression) map[string]sql.Expression {
+	var children = make(map[string]sql.Expression)
+	sql.Inspect(expr, func(expr sql.Expression) bool {
```

Comment:
i might be mistaken, but does the `false` short-circuit the walk and potentially miss multiple child expressions?  For example, this should fail, but if we are ending the walk after finding the first `GetField` we may be mistakenly OK'ing the query:
```SQL
mysql> select col1 + col2 from tab1 group by col1;
```